.changeset/giant-ravens-look.md

```diff
@@ -0,0 +1,43 @@
+---
+'astro': minor
+---
+
+Adds a new `astro:routes:resolved` hook to the Integration API, and deprecates `routes` passed to `astro:build:done`
+
+When building an integration, you can now get access to routes inside the `astro:routes:resolved` hook:
+
+```js
+const integration = () => {
+    return {
+        name: 'my-integration',
+        hooks: {
+            'astro:routes:resolved': ({ routes }) => {
+                console.log(routes)
+            }
+        }
+    }
+}
+```
+
+This hook runs before `astro:config:done`, and whenever a route changes in development.
+
+`routes` from `astro:build:done` is now deprecated. We recommend you use the new hook instead:
```

Comment:
This goes to `next`, can't we do a clean cut?

Reply:
Would break things on adapters and integrations. I think we can clean it before v5 directly but I think it would be better on a separate PR so it can be deprecated for a small period during the beta

Reply:
I'd be in favor of deprecating some properties in 5.0, and removing in 6.0. Deprecating in 5.0 beta and removing in stable feels too fast to me, especially since I think this PR will land closer to stable release